### PR TITLE
Upgrade project to edition 2018

### PR DESCRIPTION
**Pull request by weihanglo**
Just update to edition 2018 before edition 2021 coming.

### CODE REVIEW

File: curl-sys/lib.rs [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
 #[cfg(link_libnghttp2)]
 extern crate libnghttp2_sys;
 #[cfg(link_libz)]
 extern crate libz_sys;
-#[cfg(feature = "mesalink")]
-extern crate mesalink; // ensure lib is linked to
 #[cfg(link_openssl)]
 extern crate openssl_sys;
-#[cfg(windows)]
-extern crate winapi;
```
Comment (weihanglo) [inline curl-sys/lib.rs:14]: Here we preserve `extern crate` for openssl_sys, libnghttp2_sys and libz_sys for the sake of conditional linkage.

**Comment by weihanglo** [comment]
Really weird. It seems that all CI tests has passed. https://github.com/alexcrichton/curl-rust/actions/runs/805297756

File: curl-sys/lib.rs [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
 #![doc(html_root_url = "https://docs.rs/curl-sys/0.3")]
 
-extern crate libc;
+// These `extern crate` are required for conditional linkages of curl.
 #[cfg(link_libnghttp2)]
 extern crate libnghttp2_sys;
 #[cfg(link_libz)]
 extern crate libz_sys;
-#[cfg(feature = "mesalink")]
-extern crate mesalink; // ensure lib is linked to
```
Comment (alexcrichton) [inline curl-sys/lib.rs:10]: I think this needs to remain for the linkage too?

**Comment by alexcrichton** [comment]
Thanks for this!

File: curl-sys/lib.rs (hunk 2 above)
Comment (weihanglo) [reply curl-sys/lib.rs:10]: Probably not? I thought it would be linked when feature "mesalink" is on. Anyway, updated at 5768b0b.

**Pull request merged**